apps/prairielearn/src/pages/instructorStudents/components/InviteStudentsModal.tsx

```diff
@@ -0,0 +1,133 @@
+import { useMutation } from '@tanstack/react-query';
+import clsx from 'clsx';
+import { Alert, Modal } from 'react-bootstrap';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+import type { InviteResult } from '../instructorStudents.shared.js';
+
+interface InviteStudentForm {
+  uids: string;
+}
+
+export function InviteStudentsModal({
+  show,
+  onHide,
+  onSubmit,
+}: {
+  show: boolean;
+  onHide: () => void;
+  onSubmit: (uids: string[]) => Promise<InviteResult>;
+}) {
+  const {
+    register,
+    handleSubmit,
+    clearErrors,
+    reset,
+    formState: { errors },
+  } = useForm<InviteStudentForm>({
+    mode: 'onSubmit',
+    reValidateMode: 'onSubmit',
+    defaultValues: { uids: '' },
+  });
+
+  const parseUids = (value: string): string[] => {
+    return [
+      ...new Set(
+        value
+          .split(/[\n,\s]+/)
+          .map((uid) => uid.trim())
+          .filter((uid) => uid.length > 0),
+      ),
+    ];
+  };
+
+  const validateUidsFormat = (value: string): string | true => {
+    const uids = parseUids(value);
+
+    if (uids.length === 0) {
+      return 'At least one UID is required';
+    }
+
+    const invalidEmails = uids.filter((uid) => !z.string().email().safeParse(uid).success);
+
+    if (invalidEmails.length > 0) {
+      return `The following UIDs were invalid: "${invalidEmails.join('", "')}"`;
+    }
+
+    return true;
+  };
+
+  const saveMutation = useMutation({
+    mutationFn: async (uids: string[]) => {
+      return onSubmit(uids);
+    },
+    onSuccess: () => {
+      reset();
+      onHide();
+    },
+  });
+
+  const onFormSubmit = async (data: InviteStudentForm) => {
+    const uids = parseUids(data.uids);
+    saveMutation.mutate(uids);
+  };
+
+  const onExited = () => {
+    reset();
+    clearErrors();
+    saveMutation.reset();
+  };
+
+  return (
+    <Modal show={show} backdrop="static" onHide={onHide} onExited={onExited}>
+      <Modal.Header closeButton>
+        <Modal.Title>Invite students</Modal.Title>
+      </Modal.Header>
+
+      <form onSubmit={handleSubmit(onFormSubmit)}>
+        <Modal.Body>
+          {saveMutation.isError && (
+            <Alert variant="danger" dismissible onClose={() => saveMutation.reset()}>
+              {saveMutation.error instanceof Error
+                ? saveMutation.error.message
+                : 'An error occurred'}
+            </Alert>
+          )}
+          <div class="mb-0">
+            <label for="invite-uids" class="form-label">
+              UIDs
+            </label>
+            <textarea
+              id="invite-uids"
+              class={clsx('form-control', errors.uids && 'is-invalid')}
+              rows={5}
+              placeholder="One UID per line, or comma/space separated"
```

Comment:
This text belongs in help text below the input so that it doesn't disappear once one starts typing. You can add something like `student@example.com` if you want placeholder text. See https://github.com/PrairieLearn/PrairieLearn/pull/13537#discussion_r2604000316.